Stripe/STPBINRange.m

```diff
@@ -30,22 +30,30 @@ @implementation STPBINRange
     static dispatch_once_t onceToken;
     dispatch_once(&onceToken, ^{
         NSArray *ranges = @[
-                            // Catch-all values
                             @[@"", @"", @16, @(STPCardBrandUnknown)],
+
                             @[@"34", @"34", @15, @(STPCardBrandAmex)],
                             @[@"37", @"37", @15, @(STPCardBrandAmex)],
-                            @[@"30", @"30", @14, @(STPCardBrandDinersClub)],
-                            @[@"36", @"36", @14, @(STPCardBrandDinersClub)],
-                            @[@"38", @"39", @14, @(STPCardBrandDinersClub)],
-                            @[@"60", @"60", @16, @(STPCardBrandDiscover)],
-                            @[@"62", @"62", @16, @(STPCardBrandDiscover)],
-                            @[@"64", @"65", @16, @(STPCardBrandDiscover)],
+
+                            @[@"30000000", @"30599999", @14, @(STPCardBrandDinersClub)],
```

Comment:
I think we should keep these the way they were before (more permissive and matching the behavior of stripe.js), and then replace the "62" Discover range with UnionPay in #906 .

Reply:
Got it! Thanks for checking in on this.